met-web/src/services/UserServices.ts

```diff
@@ -36,24 +38,24 @@ const initKeycloak = (dispatch: Dispatch<any>) => {
         return;
       }
 
-      if (
-        !KeycloakData.resourceAccess ||
-        !KeycloakData.resourceAccess[Keycloak_Client]
-      ) {
+      if (!KeycloakData.resourceAccess) {
+        dispatch(userAuthentication(false));
         doLogout();
         return;
       }
 
       if (authenticated) {
         const UserRoles = KeycloakData.resourceAccess[Keycloak_Client].roles;
         dispatch(userRoles(UserRoles));
+
         dispatch(userToken(KeycloakData.token));
 
         KeycloakData.loadUserInfo().then((res: UserDetail) => {
           dispatch(userDetails(res));
           dispatch(userAuthorization(true));
         });
-        dispatch(userAuthentication(authenticated));
+
+        dispatch(userAuthentication(KeycloakData.authenticated ? true : false));
```

Comment:
how is KeycloakData.authenticated different that the authenticated prop passed to this method in the beginning?